sdk/src/types/block/output/basic.rs

```diff
@@ -152,8 +156,34 @@ impl BasicOutputBuilder {
         self
     }
 
+    /// Adds a storage deposit if one is needed to cover the current amount.
+    pub fn with_sufficient_storage_deposit(
```

Comment:
From the comment and function name it's a bit unclear to me why it adding a StorageDepositReturnUnlockCondition if amount < rent_cost

Not clear to me at all why we want this function

Reply:
Yes, but just reading these two lines I would assume it just increases the amount to the minimum required storage deposit and that's it, but it also adds a storage deposit return unlock condition if not enough amount is already there

Reply:
Well increasing the amount would be invalid, because that would change the ultimate transferred amount. I get what you're saying though, I suppose. Do you have a suggestion?

Reply:
`Adds a storage deposit`
Did you mean a storage deposit return unlock condition?
I think something like this would make it more clear
`If amount < rent_cost, adds a storage deposit return unlock condition for rent_cost and increases the amount to amount + rent_cost`

Reply:
Hopefully this is better now